src/lib/y2network/interfaces_collection.rb

```diff
@@ -83,10 +83,11 @@ def by_busid(busid)
 
     # Returns list of interfaces of given type
     #
-    # @param type [InterfaceType] device type
+    # @param type [InterfaceType,String,Symbol] device type or its short name
     # @return [InterfacesCollection] list of found interfaces
     def by_type(type)
-      InterfacesCollection.new(interfaces.select { |i| i.type == type })
+      short_name = type.is_a?(Y2Network::InterfaceType) ? type.short_name : type.to_s
+      InterfacesCollection.new(interfaces.select { |i| i.type.short_name == short_name })
```

Comment:
does it mean that ```==``` operator for InterfaceType is broken or that ```by_type``` just was accepting ```InterfaceType``` as an argument only? In the second case it is not fix of broken method but improvement ;-)

Reply:
mmm, good point. I have improved the implementation relying on the `Y2Network::Interfacetype.from_short_name` method.